app/services/moderator/manage_activity_and_roles.rb

```diff
@@ -99,6 +99,11 @@ def assign_elevated_role_to_user(user, role)
       check_super_admin
       remove_negative_roles
       user.add_role(role)
+
+      # Clear cache key if the elevated role matches Rack::Attack bypass roles
+      return unless Rack::Attack::ADMIN_ROLES.include?(role.to_s)
+
+      Rails.cache.delete(Rack::Attack::ADMIN_API_CACHE_KEY)
```

Comment:
When assigning elevated roles to users we check if it matches the Rack::Attack roles that bypass throttling, so that we can clear the cache and their ApiSecrets will be taken in consideration immediately without having to wait for organic cache clear